serialization_spec/serialization.py

```diff
@@ -113,13 +119,13 @@ def make_serializer_class(model, serialization_spec):
                 {'model': model, 'fields': get_fields(serialization_spec)}
             ),
             **{
-                key: SerializerLambdaField(impl=lambda value: [str(each.id) for each in getattr(value, key).all()])
+                key: SerializationSpecPluginField(RelationshipIDsPlugin(key))
```

Comment:
I'm not mega delighted that this plugin only gets created transiently here during serialization, and doesnt exist during the first, queryset pass. That's using 'plugin' for something it wasn't intended for.

How about creating this plugin during the queryset pass, eg at lines https://github.com/dabapps/django-rest-framework-serialization-spec/blob/5566ebfb3f1adea210eab68aac19a292fc38d1f8/serialization_spec/serialization.py#L200-L206

Reply:
A plugin something like this (although I couldn't make this work properly :( ):

```python
class M2MIdsPlugin(SerializationSpecPlugin):
    def __init__(self, related_model):
        self.related_model = related_model
    def modify_queryset(self, queryset):
        return queryset.prefetch_related(Prefetch(self.key, queryset= related_model.objects.only('id')))
    def get_value(self, instance):
        return [str(each.id) for each in getattr(instance, self.key).all()]
```